metomi/rose/upgrade.py

```diff
@@ -778,9 +778,35 @@ def get_meta_upgrade_module(meta_path):
     return version_module
 
 
-def parse_upgrade_args(argv=None):
+def parse_upgrade_args():
     """Parse options/arguments for rose macro and upgrade."""
-    opt_parser = metomi.rose.macro.RoseOptionParser()
+    opt_parser = metomi.rose.macro.RoseOptionParser(
+        usage='rose app-upgrade [OPTIONS] [VERSION]',
+        description='''
+Upgrade an application configuration using metadata upgrade macros.
+
+Alternatively, show the available upgrade/downgrade versions:
+
+* `=` indicates the current version.
+* `*` indicates the default version to change to.
+
+If an application contains optional configurations, loop through
+each one, combine with the main, upgrade it, and re-create it as
+a diff vs the upgraded main configuration.
+        ''',
+        epilog='''
+ARGUMENTS
+    VERSION
+        A version to change to. If no version is specified, show available
+        versions. If `--non-interactive` is used, use the latest version
+        available. If `--non-interactive` and `--downgrade` are used, use
```

Comment:
Do we need to say what happens if you just use the `--downgrade` without the `--non-interactive`?

Reply:
Feel free to do this on follow-up if you feel necessary @oliver-sanders. I thought it minor so merged anyway.